### PR TITLE
Brick utilization during sds sync

### PR DESCRIPTION
**Pull request by GowthamShanmugam**
Signed-off-by: root <root@dhcp43-187.lab.eng.blr.redhat.com>

### CODE REVIEW

File: tendrl/gluster_integration/objects/brick/brick_utilization.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
+import os
+from tendrl.commons.event import Event
+from tendrl.commons.message import Message
+from tendrl.commons.utils import cmd_utils
+ONE_GB_BYTES = 1073741824.0
+
+
+def _getMountPoint(path):
+    mount = os.path.realpath(path)
```
Comment (GowthamShanmugam) [inline tendrl/gluster_integration/objects/brick/brick_utilization.py]: this method will find mount_point from path

File: tendrl/gluster_integration/objects/brick/brick_utilization.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    for mount, info in mountPoints.iteritems():
+        mountDetail[mount] = _getStats(mount)
+        mountDetail[mount].update(
+            _getThinPoolStat(os.path.realpath(info['device']))
+        )
+        mountDetail[mount].update({'mount_point': mount})
+    return mountDetail
+
+
+def brick_utilization(path):
```
Comment (GowthamShanmugam) [inline tendrl/gluster_integration/objects/brick/brick_utilization.py]: pass brick path (e.g) /brick/brick1/b1

**Comment by GowthamShanmugam** [comment]
@r0h4n @shtripat @nnDarshan @nthomas-redhat @anmolbabu please revirew

**Comment by GowthamShanmugam** [comment]
@shtripat @nthomas-redhat @r0h4n @anmolbabu @nnDarshan please review

File: tendrl/gluster_integration/sds_sync/__init__.py [hunk 3]
```diff
@@ -150,21 +151,40 @@ def _run(self):
                                 ],
                             )
                             volume.save()
-
+                            node_context = NS.tendrl.objects.NodeContext(
+                                    node_id=NS.node_context.node_id).load()
```
Comment (r0h4n) [inline tendrl/gluster_integration/sds_sync/__init__.py]: This is not required, just do a 

NS.node_context = NS.node_context.load()

File: tendrl/gluster_integration/objects/brick/brick_utilization.py [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+         'used_percent_inode': 0.0019073486328125,
+         'thinpool_used': None,
+         'metadata_size': None,
+         'metadata_free': None,
+         'thinpool_size': None
+        }
+    }"""
+    # Above loggic will find mount_path from path
+    # Find hostname from node context
+    node_context = NS.tendrl.objects.NodeContext(
```
Comment (r0h4n) [inline tendrl/gluster_integration/objects/brick/brick_utilization.py]: NS.node_context is already present , no need to load here

File: tendrl/gluster_integration/sds_sync/__init__.py [hunk 5]
```diff
@@ -150,21 +151,40 @@ def _run(self):
...
+                                # Delete bricks only belongs to this node
+                                bricks  = NS.etcd_orm.client.read(
                                     "clusters/%s/Volumes/%s/Bricks" % (
                                         NS.tendrl_context.integration_id,
-                                        volume.vol_id,
+                                        volume.vol_id
                                     ),
                                     recursive=True
                                 )
+                                for brick in bricks.leaves:
```
Comment (nnDarshan) [inline tendrl/gluster_integration/sds_sync/__init__.py]: This will mess up the sequence

**Comment by GowthamShanmugam** [comment]
@r0h4n @nnDarshan @nthomas-redhat @anmolbabu changes done

File: tendrl/gluster_integration/objects/brick/__init__.py [hunk 6]
```diff
@@ -6,25 +6,29 @@ class Brick(objects.BaseObject):
     def __init__(
         self,
         vol_id=None,
+        sequence_id=None,
```
Comment (nnDarshan) [inline tendrl/gluster_integration/objects/brick/__init__.py]: sequence_number would be better

File: tendrl/gluster_integration/sds_sync/brick_utilization.py [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+         'thinpool_free': None,
+         'metadata_used': None,
+         'thinpool_used_percent': None,
+         'used_percent_inode': 0.0019073486328125,
+         'thinpool_used': None,
+         'metadata_size': None,
+         'metadata_free': None,
+         'thinpool_size': None
+    }"""
+    # Above loggic will find mount_path from path
```
Comment (nnDarshan) [inline tendrl/gluster_integration/sds_sync/brick_utilization.py]: below ?

**Comment by GowthamShanmugam** [comment]
@nnDarshan changes done

File: tendrl/gluster_integration/sds_sync/brick_utilization.py [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+        out = {'thinpool_size': None,
+               'thinpool_used_percent': None,
+               'metadata_size': None,
+               'metadata_used_percent': None,
+               'thinpool_free': None,
+               'metadata_free': None,
+               'thinpool_used': None,
+               'metadata_used': None}
+
+        if lvs and device in lvs and \
```
Comment (r0h4n) [inline tendrl/gluster_integration/sds_sync/brick_utilization.py]: where is this "lvs" coming from? L122? If yes, stop using a function global variable inside this function, pass it as an argument

File: tendrl/gluster_integration/objects/definition/gluster.yaml [hunk 9]
```diff
@@ -127,6 +127,9 @@ namespace.gluster:
         vol_id:
           help: Volume id
           type: String
+        sequence_number:
+          help: Sequence  id
```
Comment (r0h4n) [inline tendrl/gluster_integration/objects/definition/gluster.yaml]: what is the significance of sequence_number, add that to help

Comment (GowthamShanmugam) [inline tendrl/gluster_integration/objects/definition/gluster.yaml]: @r0h4n sry it is sequence_number

File: tendrl/gluster_integration/sds_sync/brick_utilization.py (hunk 8 above)
Comment (GowthamShanmugam) [inline tendrl/gluster_integration/sds_sync/brick_utilization.py]: no lvs is identified from https://github.com/Tendrl/gluster-integration/pull/244/files#diff-d85a92b3b059b22fbf7b2530c58f19f3R49
it just a variable inside function, if see there function definition present inside one function, so it looks like global variabl.

Comment (GowthamShanmugam) [inline tendrl/gluster_integration/sds_sync/brick_utilization.py]: Here lvm is identified https://github.com/Tendrl/gluster-integration/pull/244/files#diff-d85a92b3b059b22fbf7b2530c58f19f3R122

Comment (GowthamShanmugam) [inline tendrl/gluster_integration/sds_sync/brick_utilization.py]: Identifying lvm using executing some command from cmd_utils

File: tendrl/gluster_integration/sds_sync/brick_utilization.py [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+         'thinpool_free': None,
+         'metadata_used': None,
+         'thinpool_used_percent': None,
+         'used_percent_inode': 0.0019073486328125,
+         'thinpool_used': None,
+         'metadata_size': None,
+         'metadata_free': None,
+         'thinpool_size': None
+    }"""
+    # Below loggic will find mount_path from path
```
Comment (r0h4n) [inline tendrl/gluster_integration/sds_sync/brick_utilization.py]: typo "loggic"

File: tendrl/gluster_integration/sds_sync/brick_utilization.py [hunk 11]
```diff
@@ -0,0 +1,155 @@
+import os
+from tendrl.commons.event import Event
+from tendrl.commons.message import Message
+from tendrl.commons.utils import cmd_utils
+ONE_GB_BYTES = 1073741824.0
+
+
+def _getMountPoint(path):
```
Comment (r0h4n) [inline tendrl/gluster_integration/sds_sync/brick_utilization.py]: change to _get_mount_point, current func name is not according to PEP8

File: tendrl/gluster_integration/sds_sync/brick_utilization.py [hunk 12]
```diff
@@ -0,0 +1,155 @@
...
+    mount = os.path.realpath(path)
+    while not os.path.ismount(mount):
+        mount = os.path.dirname(mount)
+    return mount
+
+
+def _parseProcMounts(filter=True):
```
Comment (r0h4n) [inline tendrl/gluster_integration/sds_sync/brick_utilization.py]: current func name is not according to PEP8, change to small_case, same with other function names in this file

File: tendrl/gluster_integration/sds_sync/brick_utilization.py [hunk 13]
```diff
@@ -0,0 +1,155 @@
...
+    mountPoints = {}
```
Comment (r0h4n) [inline tendrl/gluster_integration/sds_sync/brick_utilization.py]: camel case not required for variables, use small_case everywhere in this file

File: tendrl/gluster_integration/sds_sync/brick_utilization.py (hunk 11 above)
Comment (GowthamShanmugam) [inline tendrl/gluster_integration/sds_sync/brick_utilization.py]: ok

File: tendrl/gluster_integration/sds_sync/brick_utilization.py (hunk 12 above)
Comment (GowthamShanmugam) [inline tendrl/gluster_integration/sds_sync/brick_utilization.py]: ok

File: tendrl/gluster_integration/sds_sync/brick_utilization.py (hunk 13 above)
Comment (GowthamShanmugam) [inline tendrl/gluster_integration/sds_sync/brick_utilization.py]: ok

File: tendrl/gluster_integration/sds_sync/brick_utilization.py (hunk 10 above)
Comment (GowthamShanmugam) [inline tendrl/gluster_integration/sds_sync/brick_utilization.py]: sry :)

**Comment by GowthamShanmugam** [comment]
@r0h4n changes done

**Pull request merged**